### PR TITLE
[gatsby-remark-autolink-headers] Add option to choose a SVG icon

### PR DESCRIPTION
**Pull request by LeKoArts**
Since v2 got merged to master I re-did my previous PR https://github.com/gatsbyjs/gatsby/pull/5926.
Oddly enough I had to delete a duplicate of the options this time 🤔

### CODE REVIEW

**Comment by pieh** [comment]
This look good, only need to fix those tests. Few should be fixed if you update your branch with latest master. `gatsby-remark-autolink-headers` specific tests would need some manual adjustments - either in tets pass empty object as second param or do check for that in plugin code before trying to destructure second argument. Maybe it would be useful to also add test there that uses custom icon?

**Comment by LeKoArts** [comment]
@pieh 

>  Few should be fixed if you update your branch with latest master.

When I created the branch it had the latest content of Gatsby. Or what do you mean?

Regarding the tests:
As written in the previous PR I'm not proficient enough with tests yet. I'd either need a helping hand or someone else to do the changes :)

**Comment by pieh** [comment]
> When I created the branch it had the latest content of Gatsby. Or what do you mean?

Yeah, this one is recent that is not in your branch https://github.com/gatsbyjs/gatsby/commit/56e94fb68c3d83b449aa5ea33276eea4fad3006c that will fix contentful plugin tests failing in travis

> As written in the previous PR I'm not proficient enough with tests yet. I'd either need a helping hand or someone else to do the changes :)

Let's try helping hand then - https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-autolink-headers/src/__tests__/index.js - You would to adjust lines 16 and 34:
```diff
-const transformed = plugin({ markdownAST })
+const transformed = plugin({ markdownAST }. {})
```

Adding new test for icon: You can copy previous and adjust `plugin` call to include `icon` - this will generate updated snapshot, which you'll need to add to PR

**Comment by LeKoArts** [comment]
@pieh Thanks for the helping hand! The test for the commit "First step with tests" were successful. Let's see if I added the tests for the icon usage the right way!

File: packages/gatsby-remark-autolink-headers/README.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ module.exports = {
 ## Options
 
 - `offsetY`: Signed integer, vertical offset value in pixels, e.g.
+- `icon`: Use your own svg icon
```
Comment (pieh) [inline packages/gatsby-remark-autolink-headers/README.md]: Maybe worth mentioning that both options are optional? Other than maybe this, this PR is ready to go in

Comment (LeKoArts) [reply packages/gatsby-remark-autolink-headers/README.md]: Did update the README 👍

**Comment by LeKoArts** [comment]
@pieh When does it get published to npm for usage?

**Comment by pieh** [comment]
@LekoArts When @KyleAMathews or @m-allanson will have time to do new release (can't give you exact time). Unfortunately I can't do releases because `yarn pack` that is used to do that is broken on my OS (Windows) and will cause faulty releases ( https://github.com/yarnpkg/yarn/pull/5979 )

**Comment by m-allanson** [comment]
@LeKoArts this has been released as `gatsby-remark-autolink-headers@2.0.0-beta.2`

**Comment by brandonkal** [comment]
@m-allanson  Is there a reason this is labeled beta? What part needs testing?

**Comment by KyleAMathews** [comment]
This is part of work for v2 — core Gatsby and all plugins are in beta right now.

**Pull request merged**